openmls/src/treesync/mod.rs

```diff
@@ -161,19 +165,27 @@ impl TreeSync {
         sender_kpr: &KeyPackageRef,
         path_secret_option: impl Into<Option<PathSecret>>,
         key_package_bundle: KeyPackageBundle,
-    ) -> Result<(Self, Option<CommitSecret>), TreeSyncError> {
+    ) -> Result<(Self, Option<CommitSecret>), TreeSyncFromNodesError> {
         let mut tree_sync =
             Self::from_nodes(backend, ciphersuite, node_options, key_package_bundle)?;
 
         // Get the leaf index of the sender.
-        let sender_index = tree_sync.leaf_index(sender_kpr)?;
+        let sender_index = tree_sync
+            .leaf_index(sender_kpr)
+            .map_err(|_| LibraryError::custom("Sender not in tree"))?;
```

Comment:
Why is this a `LibraryError`? Similar to some of the other comments, I don't think we should start making assumptions about inputs by throwing `LibraryErrors` if they are malformed..